charts/thanos/templates/query-frontend/pdb.yaml

```
@@ -1,4 +1,4 @@
-{{- if and .Values.queryFrontend.enabled .Values.queryFrontend.podDisruptionBudget.enabled -}}
+{{- if and .Values.query.enabled .Values.queryFrontend.enabled .Values.queryFrontend.podDisruptionBudget.enabled -}}
```

```suggestion
{{- if and .Values.query.enabled (and .Values.queryFrontend.enabled .Values.queryFrontend.podDisruptionBudget.enabled) -}}
```
Query disabled should logically short circuit other checks.